content/docs/hoon/guides/sail.md

```diff
@@ -0,0 +1,495 @@
+represent an XML hierarchical structure with a single root node. There are six
+XML-related types defined in the standard library:
+
+```hoon
++$  mane  $@(@tas [@tas @tas])                    ::  XML name+space
++$  manx  $~([[%$ ~] ~] [g=marx c=marl])          ::  dynamic XML node
++$  marl  (list manx)                             ::  XML node list
++$  mars  [t=[n=%$ a=[i=[n=%$ v=tape] t=~]] c=~]  ::  XML cdata
++$  mart  (list [n=mane v=tape])                  ::  XML attributes
++$  marx  $~([%$ ~] [n=mane a=mart])              ::  dynamic XML tag
+```
+
+More information about these can be found in [section 5e of the standard library
+reference](/docs/hoon/reference/stdlib/5e).
+
+You don't need to understand these types in order to write Sail. The main thing
+to note is that a `$manx` is a node (a single tag) and its contents is a
+[$marl](/docs/hoon/reference/stdlib/5e#marl), which is just a `(list manx)`.
+
+### Rendering
+
+A `$manx` can be rendered as HTML in a tape with the `++en-xml:html` function in
+`zuse.hoon`. For example:
+
+```
+> ;p: foobar
+[[%p ~] [[%$ [%$ "foobar"] ~] ~] ~]
+
+> =x ;p: foobar
+
+> (en-xml:html x)
+"<p>foobar</p>"
+
+> (crip (en-xml:html x))
+'<p>foobar</p>'
+```
+
+You may wish to do this in your agent directly, but you can also leave mark
+conversion to Eyre.
+
+### Sanitization
+
+The `++en-xml:html` function will sanitize the contents of both attributes and
+elements, converting characters such as `>` to HTML entities. For example:
+
+```
+> =z ;p(class "\"><script src=\"example.com/xxx.js"): <h1>FOO</h1>
+
+> (crip (en-xml:html z))
+'<p class="&quot;&gt;&lt;script src=&quot;example.com/xxx.js">&lt;h1&gt;FOO&lt;/h1&gt;</p>'
+```
+
+### Marks
+
+The mark for a raw `$manx` is a `%hymn`. This mark is included in the `%base`
+desk. The mark for a rendered HTML cord is `%html`. The `%hymn` mark includes
+conversion methods to both `%html` and `%mime`. The `%html` mark includes
+conversion methods to `%mime` and `%hymn`.
+
+If you want to produce an HTML document through Eyre's scry interface, for
+example, you can either produce a `$manx` with a `%hymn` mark and let Eyre
+convert it to `%html`, or else you can call `++en-xml:html` and then `++crip` on
+the product of your Sail expressions and produce a cord with an `%html` mark
+directly.
+
+## Sail Runes
+
+In addition to the syntax so far described, there are also a few Sail-specific
+runes:
+
+### `;+` Miclus
+
+The [miclus rune](/docs/hoon/reference/rune/mic#-miclus) makes a `$marl` from a
+complex hoon expression that produces a single `$manx`. Its main use is nesting
+tall-form hoon logic in another Sail element. For example:
+
+```hoon
+;p
+  ;b: {(a-co:co number)}
+  ; is an
+  ;+  ?:  =(0 (mod number 2))
+        ;b: even
+      ;b: odd
+  ; number.
+==
+```
+
+Produces one of these depending on the value of `number`:
+
+```
+<p><b>2 </b>is an <b>even </b>number.</p>
+```
+
+```
+<p><b>12345 </b>is an <b>odd </b>number.</p>
+```
+
+### `;/` Micfas
```

Comment:
I would move this one to be the last rune mentioned, just because the other ones are so similar to each other, this one kinda stands out.